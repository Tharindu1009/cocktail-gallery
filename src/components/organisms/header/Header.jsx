import './styles/header.scss';
import { Typography } from '@mui/material';
import { useNavigate } from 'react-router-dom';

function Header() {
    const navigate = useNavigate();

    const goToHome = () => {
        // redirect to the home page
        navigate("/");
    }

    return (
        <div data-testid="header" className='header'>
            <Typography data-testid="headerTitle" variant="h6" className='title' onClick={goToHome}>
                Cocktail Gallery
            </Typography>
        </div>
    )
}

export default Header;