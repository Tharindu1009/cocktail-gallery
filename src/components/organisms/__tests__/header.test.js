import { render, cleanup, fireEvent } from "@testing-library/react";
import TestRenderer from 'react-test-renderer';
import Header from "../header/Header";
import { Typography } from '@mui/material';

jest.mock('react-router-dom', () => ({
    useNavigate: jest.fn(),
}));

describe('Header', () => {
    afterEach(() => {
        cleanup();
    })

    const mockGoToHome = jest.fn();

    it('should render header component correctly', () => {
        const { getByTestId } = render(<Header />);
        const headerElement = getByTestId('header');
        expect(headerElement).toBeInTheDocument();
        expect(headerElement).toHaveTextContent('Cocktail Gallery');
    });

    it('clicking header title and then should navigates to home', () => {
        const { getByTestId } = render(<Typography variant="h6" data-testid="headerTitle" className='title'
            onClick={mockGoToHome}> Cocktail Gallery </Typography>);
        const headerTitleElement = getByTestId('headerTitle');
        fireEvent.click(headerTitleElement);
        expect(mockGoToHome).toHaveBeenCalledTimes(1);
    });

    it('matches header snapshot', () => {
        const tree = TestRenderer.create(<Header />).toJSON();
        expect(tree).toMatchSnapshot();
    })
});