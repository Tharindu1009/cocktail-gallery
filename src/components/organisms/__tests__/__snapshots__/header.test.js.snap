// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches alert snapshot 1`] = `
<div
  className="header"
  data-testid="header"
>
  <h6
    className="MuiTypography-root MuiTypography-h6 title css-2ulfj5-MuiTypography-root"
    onClick={[Function]}
  >
    Cocktail Gallery
  </h6>
</div>
`;

exports[`Header matches header snapshot 1`] = `
<div
  className="header"
  data-testid="header"
>
  <h6
    className="MuiTypography-root MuiTypography-h6 title css-2ulfj5-MuiTypography-root"
    data-testid="headerTitle"
    onClick={[Function]}
  >
    Cocktail Gallery
  </h6>
</div>
`;
